packages/caravan-fees/src/btcTransactionTemplate.ts

```diff
@@ -0,0 +1,506 @@
+  }
+
+  /**
+   * Calculates the total change amount. (Total Inputs Amount - Total Non-change (non-malleable) Outputs Amount )
+   * @returns {Satoshis} The total change amount in satoshis (as a string)
+   */
+  get changeAmount(): Satoshis {
+    return this.calculateChangeAmount().toString();
+  }
+
+  /**
+   * Calculates the total output amount.
+   * @returns {Satoshis} The total output amount in satoshis (as a string)
+   */
+  getTotalOutputAmount(): Satoshis {
+    return this.calculateTotalOutputAmount().toString();
+  }
+
+  /**
+   * Estimates the virtual size of the transaction.
+   * @returns The estimated virtual size in vbytes
+   */
+  get estimatedVsize(): number {
+    return estimateTransactionVsize({
+      addressType: this._scriptType,
+      numInputs: this._inputs.length,
+      numOutputs: this._outputs.length,
+      m: this._requiredSigners,
+      n: this._totalSigners,
+    });
+  }
+
+  /**
+   * Calculates the estimated fee rate of the transaction.
+   * @returns {string} The estimated fee rate in satoshis per vbyte
+   */
+  get estimatedFeeRate(): string {
+    return this.calculateFee()
+      .dividedBy(this.calculateEstimatedVsize())
+      .toString();
+  }
+
+  /**
+   * Adds an input to the transaction.
+   * @param input - The input to add
+   */
+  addInput(input: BtcTxInputTemplate): void {
+    this._inputs.push(input);
+  }
+
+  /**
+   * Adds an output to the transaction.
+   * @param output - The output to add
+   */
+  addOutput(output: BtcTxOutputTemplate): void {
+    this._outputs.push(output);
+  }
+
+  /**
+   * Removes an output from the transaction.
+   * @param index - The index of the output to remove
+   */
+  removeOutput(index: number): void {
+    this._outputs.splice(index, 1);
+  }
+
+  /**
+   * Adjusts the change output of the transaction.
+   * This method calculates a new change amount based on the current inputs,
+   * non-change outputs, and the target fee. It then updates the change output
+   * or removes it if the new amount is below the dust threshold.
+   *
+   * The method also handles cases where the change output already has an amount:
+   * 1. It calculates the difference between the new and current change amount.
+   * 2. If removing change, it ensures the fee doesn't unexpectedly increase.
+   */
+  adjustChangeOutput(): void {
+    if (this.malleableOutputs.length === 0) return;
+
+    // TO DO (MRIGESH):
+    // To handle for multiple change outputs
+    //
+    const changeOutput = this.malleableOutputs[0];
+    const totalOutWithoutChange = this.calculateTotalOutputAmount().minus(
+      this.calculateChangeAmount(),
+    );
+    const currentChangeAmount = new BigNumber(changeOutput.amountSats);
+
+    const newChangeAmount = this.calculateTotalInputAmount()
+      .minus(totalOutWithoutChange)
+      .minus(new BigNumber(this.targetFeesToPay));
+
+    if (newChangeAmount.gte(0)) {
+      const changeAmountDiff = newChangeAmount.minus(currentChangeAmount);
```

Comment:
What if we just set a new amount rather than coming up with a diff and setting up the difference?

Reply:
I just added it to check that if we actually need to change the amount of the changeOutput ... if we have the new change same as the old one ?